handler.go

```diff
@@ -342,3 +310,51 @@ func closeSendStream(ctx context.Context, stream *yarpctransport.ClientStream) e
 	}
 	return nil
 }
+
+// streamIOHandler uses provided stream message reader to provide stream
+// IO methods to read requests one by one and handle responses.
+type streamIOHandler struct {
```

Comment:
As talked offline, let's call this struct `streamIOInitializer`. This would match with other APIs using this struct (like `shouldMakeInitialRequest` for instance).